xbmc/music/windows/GUIWindowMusicBase.cpp

```diff
@@ -572,7 +572,8 @@ void CGUIWindowMusicBase::GetContextButtons(int itemNumber, CContextButtons &but
 
     if (!item->IsParentFolder())
     {
-      if (item->CanQueue() && !item->IsAddonsPath() && !item->IsScript())
+      if (item->CanQueue() && !item->IsAddonsPath() && !item->IsScript() &&
```

Comment:
imho function `CanQueue()` of CFileItem needs to be fixed so it can be the only condition of this if statement?
sadly i‘ve no idea if this would open a can

Reply:
That's a nice point - I also don't like the current implementation (there's even a comment to remove hacks based on those properties.
I think for now I'll fix the CanQueue condition for addon items and leave a todo to remove the other conditions later